Slider/Assets/Scripts/Map/Ocean/BottleManager.cs

```diff
@@ -0,0 +1,124 @@
+{
+    // set in the editor
+    public GameObject bottle_prefab; 
+
+    private int turncounter = 0;
+    private GameObject bottle = null;
+    private bool puzzleActive = false;
+    private bool puzzleSolved = false;
+    private STile boundSTile = null;
+
+    private List<Vector3> positions = new List<Vector3>{ new Vector3(-4.5f,7.5f), new Vector3(0,0), new Vector3(4.5f,-7.5f)};
+
+    public void OnEnable()
+    {
+        //subscribe stile move end to update counter
+        SGridAnimator.OnSTileMoveStart += UpdateBottleLocation;
+    }
+
+
+    public void OnDisable()
+    {
+        //unsubscribe stile move end to update counter
+        SGridAnimator.OnSTileMoveStart -= UpdateBottleLocation;
+    }
+
+    protected IEnumerator StartBottleMovementAnimation(Vector3 start, Vector3 end, float moveDuration)
+    {
+        float t = 0;
+        while(t < moveDuration)
+        {
+            t += Time.deltaTime;
+            bottle.transform.localPosition = Vector3.Lerp(start, end, t/moveDuration);
+            yield return null;
+        }
+    }
+
+    private void UpdateBottleLocation(object sender, SGridAnimator.OnTileMoveArgs tileMoveArgs){
+        if(puzzleActive && tileMoveArgs.stile.islandId == boundSTile.islandId)
+        {
+            turncounter++;
+            if (turncounter >2)
+            {
+                DestroyBottle();
+            }
+            else
+                StartCoroutine(StartBottleMovementAnimation(positions[turncounter-1] ,positions[turncounter], tileMoveArgs.moveDuration));
+        }
+    }
+
+    private void DestroyBottle()
+    {
+        Debug.Log("deleting bottle");
+        if(bottle != null)
+            bottle.GetComponent<Animator>().SetBool("IsSinking", true);
+        bottle = null;
+        puzzleActive = false;
+        boundSTile = null;
+        
+    }
+
+    public void CreateNewBottle()
+    {
+        //create new bottle after talking to romeo and puzzle is not solved
+        if(bottle == null && !puzzleSolved)
+        {
+            puzzleActive = true;
+
+            Debug.Log("Creating new bottle");
+            bottle = GameObject.Instantiate(bottle_prefab, new Vector3(-7.5f,41.5f), Quaternion.identity);
+
+            boundSTile = SGrid.GetSTileUnderneath(bottle.transform, null);
+            Debug.Log("bound stile: " + boundSTile);
+            bottle.transform.SetParent(boundSTile.transform);
+
+            if(boundSTile.islandId == 1 || boundSTile.islandId == 3 || boundSTile.islandId == 8)
+            {
+                turncounter = 0;
+                bottle.transform.localPosition = positions[turncounter];
+                UITrackerManager.AddNewTracker(bottle, UITrackerManager.DefaultSprites.circleEmpty);
+                Debug.Log("valid tile");
+            }
+            else
+            {
+                Debug.Log("invalid tile");
+                DestroyBottle();
+            }
+
+
+            
+        }
+    }
+
+    public void InvalidTile(Condition c)
```

Comment:
Is this needed/used? NPCs have a dcond that specifically checks grid regex alr

Reply:
That's true but its convenient if we want to change the set of valid tiles later (now that i made a string with all the valid tiles)